Aagam_Shah/postgreAssignment/src/controller/student.controller.ts

```diff
@@ -0,0 +1,216 @@
+    try{
+      const response = await db.query(
+        'select * from school.student where classid = $1', [id]
+      );
+      if(response.rows.length){
+        ctx.response.status = 200;
+        ctx.body = response.rows;
+        return;
+      }
+      ctx.response.status = 400;
+      ctx.body = "Required Id is not Present";
+      return;
+    }
+    catch(error){
+      ctx.response.status = 500;
+      ctx.body = error.message;
+      return;
+    }
+  }
+  else{
+    ctx.body = "invalid id passed";
+    ctx.response.status = 400;
+    return;
+  }
+};
+
+exports.studentOfTeacher = async (ctx: Context) => {
+  var id = ctx.params.id;
+  if(validation.isValidId(id)){
+    try{
+      const response = await db.query(
+        `select * 
+        from school.student as st, school.subject as sb
+        where st.classid = sb.classid and sb.teacherid =  $1`, [id]
+      );
+      if(response.rows.length){
+        ctx.response.status = 200;
+        ctx.body = response.rows;
+        return;
+      }
+      ctx.response.status = 400;
+      ctx.body = "Required Id is not Present";
+      return;
+    }
+    catch(error){
+      ctx.response.status = 500;
+      ctx.body = error.message;
+      return;
+    }
+  }
+  else{
+    ctx.body = "invalid id passed";
+    ctx.response.status = 400;
+    return;
+  }
+};
+
+exports.studentOfSubject = async (ctx: Context) => {
+  var id = ctx.params.id;
+  if(validation.isValidId(id)){
+    try{
+      const response = await db.query(
+        `select * 
+        from school.student as st, school.subject as sb
+        where st.classid = sb.classid and sb.subjectid = $1`, [id]
+      );
+      if(response.rows.length){
+        ctx.response.status = 200;
+        ctx.body = response.rows;
+        return;
+      }
+      ctx.response.status = 400;
+      ctx.body = "Required Id is not Present";
+      return;
+    }
+    catch(error){
+      ctx.response.status = 500;
+      ctx.body = error.message;
+      return;
+    }
+  }
+  else{
+    ctx.body = "invalid id passed";
+    ctx.response.status = 400;
+    return;
+  }
+};
+
+exports.topStudents = async (ctx: Context) => {
+  var number = parseInt(ctx.params.number);
+  if(typeof number === "number"){
+    try{
+      const response = await db.query(
+        `select studentid, sum(marks) as total 
+        from school.marks 
+        group by studentid 
+        order by total desc 
+        limit $1`, [number]
+      );
+      if(response.rows.length){
+        ctx.response.status = 200;
+        ctx.body = response.rows;
+        return;
+      }
+      ctx.response.status = 400;
+      ctx.body = "Required Id is not Present";
+      return;
+    }
+    catch(error){
+      ctx.response.status = 500;
+      ctx.body = error.message;
+      return;
+    }
+  }
+  else{
+    ctx.body = "invalid number passed";
+    ctx.response.status = 400;
+    return;
+  }
+};
+
+exports.createStudent = async (ctx: Context) => {
```

Comment:
is it duplicated

Reply:
yes its there in school controller also. please check all controllers and do self review once.

Reply:
removed old combined controller file